helixbass/tree-sitter-lint
# Rule macro

In this PR:
- add a `rule!()` proc-macro to make writing rules more ergonomic
- add a `violation!()` macro for arguably making declaring violations more ergonomic

To test:
Per the hard-coded/test rules you should be able to use the `rule!()` macro to declare rules that are "simple" or that have "rule-static", "per-run" and/or "per-file-run" "state"

Based on `test-single-fix`
